geometry/shapes.py

```diff
@@ -432,3 +432,45 @@ def intersects(obj1, obj2):
             return False
 
     return True
+
+
+class Triangle(Polyhedron):
+    def __init__(self, *args):
+        super(Triangle, self).__init__()
+
+        self._vertices = []
+        input_shape = np.array(args[0]).shape
+
+        if len(args) != 3:
```

Comment:
Each arg is checked if it is legal by a new added function utils.make_legal_point(pt, input_shape)